library/core/functions.error.php

```diff
@@ -692,6 +695,49 @@ function cleanErrorArguments(&$var, $blackList = ['configuration', 'config', 'da
     }
 }
 
+if (!function_exists('__cleanErrorArguments')) {
+    /**
+     * This is an internal function not to be used outside of error printing.
+     *
+     * @param $var
+     * @param array $blackList
+     */
+    function __cleanErrorArguments($var, $blackList = ['configuration', 'config', 'database', 'password']) {
+        $seen = [];
+
+        $fn = function ($var, int $nest = 0) use (&$fn, $blackList, &$seen) {
```

Comment:
Do we normally allow variable shadowing like this? It makes things a little bit harder to follow?